src/move/visualization/dataset_distributions.py

```diff
@@ -0,0 +1,262 @@
+
+    return fig
+
+
+def plot_reconstruction_movement(
+    baseline_recon, perturb_recon, k
+) -> matplotlib.figure.Figure:
+    """
+    Plot, for each sample, the change in value from the unperturbed reconstruction to the perturbed reconstruction.
+    Blue lines are left/negative shifts, red lines are right/positive shifts.
+
+    Args:
+        baseline_recon: baseline reconstruction array with s samples and k features (s,k).
+        perturb_recon:  perturbed "                                                      "
+    """
+    # Feature changes
+    fig = plt.figure(figsize=(25, 25))
+    for s in range(np.shape(baseline_recon)[0]):
+        plt.arrow(
+            baseline_recon[s, k],
+            s / 100,
+            perturb_recon[s, k],
+            0,
+            length_includes_head=True,
+            color=["r" if baseline_recon[s, k] < perturb_recon[s, k] else "b"][0],
+        )
+    plt.ylabel("Sample (e2)", size=40)
+    plt.xlabel("Feature_value", size=40)
+    return fig
+
+
+def plot_cumulative_distributions(
+    edges, hist_base, hist_pert, title
+) -> matplotlib.figure.Figure:
+
+    # Cumulative distribution:
+    fig = plt.figure(figsize=(7, 7))
+    plt.plot(
+        (edges[:-1] + edges[1:]) / 2,
+        np.cumsum(hist_base),
+        color="blue",
+        label="baseline",
+        alpha=0.5,
+    )
+    plt.plot(
+        (edges[:-1] + edges[1:]) / 2,
+        np.cumsum(hist_pert),
+        color="red",
+        label=f"Perturbed",
+        alpha=0.5,
+    )
+    # plt.plot(edges_f[:-1],hist_base_f, color="darkblue", label="baseline f", alpha=.5)
+    # plt.plot(edges_f[:-1],hist_pert_f, color="darkred", label=f"Perturbed {i} reconstruct feat_{j} f", alpha=.5)
+    plt.title(f"{title}.png")
+    plt.xlabel("Feature value")
+    plt.ylabel("Cumulative distribution")
+    plt.legend()
+
+    return fig
+
+
+def plot_correlations(x, y, x_pol, y_pol, a2, a1, a, k):
+
+    # Plot correlations
+    fig = plt.figure(figsize=(3, 3))
+    plt.plot(x, y, marker=".", lw=0, markersize=1, color="red")
+    # plt.plot(x,y_2, marker='.', lw=0, markersize=1, color='k', alpha=.3)
+    plt.plot(
+        x_pol,
+        y_pol,
+        color="blue",
+        label="{0:.2f}x^2 {1:.2f}x {2:.2f}".format(a2, a1, a),
+        lw=1,
+    )
+    plt.plot(x_pol, x_pol, lw=1, color="k")
+    plt.xlabel(f"Feature {k} baseline values ")
+    plt.ylabel(f"Feature {k} baseline  value reconstruction")
+    plt.legend()
+
+    return fig
+
+
+def get_2nd_order_polynomial(x_array, y_array, n_points=100):
+    """
+    Given a set of x an y values, find the 2nd oder polynomial fitting best the data.
+    Returns:
+        x_pol: x coordinates for the polynomial function evaluation.
+        y_pol: y coordinates for the polynomial function evaluation.
+    """
+    a2, a1, a = np.polyfit(x_array, y_array, deg=2)
+
+    x_pol = np.linspace(np.min(x_array), np.max(x_array), n_points)
+    y_pol = np.array([a2 * x * x + a1 * x + a for x in x_pol])
```

Comment:
No need to do list comprehension, it's more efficient to do the vectorized operation:

```python
>>> %timeit np.array([a2 * x * x + a1 * x + a for x in x_pol])
37.3 µs ± 777 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
>>> %timeit a2 * x_pol ** 2 + a1 * x_pol + a
3.6 µs ± 43.4 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```